lib/ClangImporter/ClangAdapter.h

```diff
@@ -60,6 +60,16 @@ struct PlatformAvailability;
 Optional<const clang::Decl *>
 getDefinitionForClangTypeDecl(const clang::Decl *D);
 
+/// Returns the first redeclaration of \p D outside of a function.
+///
+/// If there are no such redeclarations, returns null.
+/// If \p D is not redeclarable, just returns \p D itself.
+///
+/// Use this similarly to clang::Decl::getCanonicalDecl when the containing
+/// module matters.
+const clang::Decl *
+getFirstNonLocalDecl(const clang::Decl *D);
```

Comment:
Does "NonLocal" == "top-level"? Or is there a case where there's a difference between the two? In case, they're the same, it might be more understandable to rename the function to `getFirstTopLevelDecl` (at first glance, I couldn't tell what "nonlocal" mean).

I think it would be helpful to provide a little bit of information from the commit message here in the doc comment. For example, the reader might not know that C even allows declarations inside functions, and that clang's "canonical" location considers those, which is why it is not appropriate here.